daisy_workflows/image_build/install_package/cos/package_replacement/compile_debian_package.sh

```diff
@@ -26,19 +26,20 @@
 #
 # Most of the code here is sourced from: https://github.com/GoogleCloudPlatform/guest-test-infra/blob/master/packagebuild/daisy_startupscript_deb.sh
 #
-# Args: ./compile_debian_package [overlays_branch] [guest_agent_version]
-# Example: ./compile_debian_package release-R113 094ef227ddf92165abcb7b1241ca44728c3086d1
-#     $1 [overlays_branch]: the COS milestone version (to apply the correct patches).
+# Args: ./compile_debian_package [guest_agent_version]
+# Example: ./compile_debian_package 094ef227ddf92165abcb7b1241ca44728c3086d1
 #     $2 [commit_sha]: the guest agent commit sha (to upgrade to).
 
 set -o errexit
 set -o pipefail
 set -o nounset
 
 apply_patches() {
-  # Download the repositories and apply COS specific patches.
+  # Download the repositories and apply COS specific patches. NOTE: The patches will be
+  # pulled from the master branch in COS. This is based on the assumption that master
+  # will have the most recent guest agent version, and therefore the most recent patches.
   echo -e "\nATTENTION: Downloading the board-overlays and guest-agent repos...\n"
-  git clone https://cos.googlesource.com/cos/overlays/board-overlays --branch ${overlays_branch}
+  git clone https://cos.googlesource.com/cos/overlays/board-overlays --branch master
```

Comment:
What if the new agent diverges so much that these patches no longer apply?

Reply:
The goal is that we keep the patches required to a minimum. IIUC, there is only one patch still needed for COS: create host key patch. The homedir gid patch was already upstreamed. So if there are significant changes for create host key, it would be nice if we knew so we could update the patches too. What do you think?

Current patch set for reference:
https://cos.googlesource.com/cos/overlays/board-overlays/+/refs/heads/master/project-lakitu/app-admin/google-guest-agent/files/

Reply:
I'm thinking that this will definitely break when we publish the refactor of the guest-agent we are working on because the directory structure will change if nothing else.

When that happens we will have two options:
1) disable COS again (kind of defeats the purpose of doing this)
2) create some way to get a different set of patches (or no patches) for upcoming changes to guest-agent. Branches seem like a good solution for this.

Reply:
I see, do you mind expanding on the branch solution?

I also remember Leo mentioning, "in this case it's non upstreamable (we can work on adding an extra configuration with the "right semantics" and have one override the other)", is this a possible solution to avoid having this patch?

Reply:
If we discover that some version of the agent in the future needs a patch which can't be upstreamed, or that we need to drop a patch from the next version but not the current version of the agent, COS could provide a `guest-agent-next` branch or something with a new patchset. If there are better solutions for this that's fine too.

I don't think the specifics of the patch are important, because you also use that directory to store your distro configuration which might need new options if we add modules that COS wants to disable or something.
It's more that there should be a way to test the guest-agent using future packaging for COS without needing to affect current COS.

Reply:
Discussed out of band, we can return to this when it's needed.